lib/tlog/sink.c

```diff
@@ -135,6 +135,20 @@ tlog_sink_flush(struct tlog_sink *sink)
     return grc;
 }
 
+void
+tlog_sink_io_close(struct tlog_sink *sink,
+                bool output)
+{
+    assert(sink == NULL || tlog_sink_is_valid(sink));
+
+    if (sink == NULL) {
+        return;
+    }
+    if (sink->type->cleanup != NULL) {
```

Comment:
You probably meant `sink->type->close != NULL` here. I would say `sink->type->io_close != NULL` would be better still :)